src/dotnet/APIView/ClientSPA/src/app/_models/azureEngSemanticVersion.ts

```diff
@@ -0,0 +1,105 @@
+    buildNumber : string = '';
+    prereleaseNumber : number = 0;
+    isPrerelease : boolean = false;
+    versionType : string = '';
+    rawVersion : string;
+    isSemVerFormat : boolean;
+    defaultPrereleaseLabel : string = '';
+    defaultAlphaReleaseLabel : string = '';
+
+    constructor(version: string, language: string) {
+        const versionParts = AzureEngSemanticVersion.SEM_VAR_REGEX.exec(version);
+        
+        if (versionParts) {
+            this.isSemVerFormat = true;
+            this.rawVersion = version;
+
+            this.major = parseInt(versionParts.groups!['major']);
+            this.minor = parseInt(versionParts.groups!['minor']);
+            this.patch = parseInt(versionParts.groups!['patch']);
+
+            if (language === "Python") {
+                this.setupPythonConventions();
+
+            } else {
+                this.setupDefaultConventions();
+            }
+
+            if (versionParts.groups!['prelabel']) {
+                this.prereleaseLabel = versionParts.groups!['prelabel']
+                this.prereleaseLabelSeparator = versionParts.groups!["presep"]
+                this.prereleaseNumber = versionParts.groups!["prenumber"] as unknown as number
+                this.prereleaseNumberSeparator = versionParts.groups!["prenumsep"]
+                this.isPrerelease = true
+                this.versionType = "Beta"
+                this.buildNumberSeparator = versionParts.groups!["buildnumsep"]
+                this.buildNumber = versionParts.groups!["buildnumber"]
+            } else {
+                // artifically provide these values for non-prereleases to enable easy sorting of them later than prereleases.
+                this.prereleaseLabel = "zzz"
+                this.prereleaseNumber = 99999999
+                this.isPrerelease = false
+                this.versionType = "GA"
+                if (this.major == 0) {
+                   // Treat initial 0 versions as a prerelease beta's
+                  this.versionType = "Beta"
+                  this.isPrerelease = true
+                }
+                else if (this.patch !== 0) {
+                  this.versionType = "Patch"
+                }
+            }
+        } else {
+            this.rawVersion = version
+            this.isSemVerFormat = false
+        }
+    }
+
+    private setupPythonConventions() : void {
+        this.prereleaseLabelSeparator = this.prereleaseNumberSeparator = this.buildNumberSeparator = ""
+        this.defaultPrereleaseLabel = "b"
+        this.defaultAlphaReleaseLabel = "a"
+    }
+
+    private setupDefaultConventions() : void {
+        this.prereleaseLabelSeparator = "-"
+        this.prereleaseNumberSeparator = "."
+        this.buildNumberSeparator = "."
+        this.defaultPrereleaseLabel = "beta"
+        this.defaultAlphaReleaseLabel = "alpha"
+    }
+
+    compareTo(other: AzureEngSemanticVersion): number {
+        if (!(other instanceof AzureEngSemanticVersion)) {
```

Comment:
Can we use npm package `semver` to compare the versions? It should also help us sort.

Reply:
Do we really want to use SemVer. I think Wes's logic provides more customization.

Reply:
@weshaggard for recommendations here

Reply:
Talked to Wes. Apparently npm SemVar does not support our python prerelease. So I think we are still better off with this logic since we will eventually need to support Python

Reply:
If we need to support sorting of prereleases then yes unfortunately the npm semver package doesn't handle python prerelease versions. However if we don't care about prereleases then I would suggest using the npm semver package and just ignore parse failures on those verisons.

Reply:
@chidozieononiwu could you please add tests to identify when a package is GAd for all languages so can validate all the language formats of their betas and make sure our logic is good?

Reply:
PS: love that we are adding tests :)

Reply:
Added tests